test/logger_test.coffee

```diff
@@ -0,0 +1,205 @@
+kv  = require "../lib/kayvee"
+logger = require "../lib/logger/logger"
+assert = require 'assert'
+_ = require 'underscore'
+_.mixin = require 'underscore.deep'
+fs = require 'fs'
+
+sample = ""
+outputFunc = (text) -> sample = text
+
+describe 'logger_test', ->
+  logObj = {}
+  logObj2 = {}
+  beforeEach ->
+  	logObj = new logger("logger-tester")
+  	sample = ""
+  	logObj.setOutput outputFunc
+
+  describe '.constructor', ->
+    it 'constructor', ->
+      formatter = (data) -> data["level"] + "." + data["source"] + "." + data["title"]
+      logObj = new logger("logger-test", logger.Info, formatter, outputFunc)
+      logObj.debug "testlogdebug"
+      expected = ""
+      assert.equal sample, expected
+
+      logObj.info "testloginfo"
+      expected = logger.Info + ".logger-test.testloginfo"
+      assert.equal sample, expected
+  describe '.validatelog', ->
+    # Explicit validation checks
+    it 'validatelog', ->
+      logLvl = logObj._validateLogLvl "debug"
+      assert.equal logLvl, logger.Debug
+      logLvl = logObj._validateLogLvl "Debug"
+      assert.equal logLvl, logger.Debug
+
+      logLvl = logObj._validateLogLvl "info"
+      assert.equal logLvl, logger.Info
+      logLvl = logObj._validateLogLvl "Info"
+      assert.equal logLvl, logger.Info
+
+      logLvl = logObj._validateLogLvl "sometest"
+      assert.equal logLvl, logger.Debug
```

Comment:
To make these test cases more explicit/readable, I might consider a pattern like:

```
describe '.validatelog', ->
   it 'is case-insensitive in log level name', ->
      logLvl = logObj._validateLogLvl "debug"
      assert.equal logLvl, logger.Debug
      logLvl = logObj._validateLogLvl "Debug"
      assert.equal logLvl, logger.Debug

  it 'sets non-default log levels', ->
      logLvl = logObj._validateLogLvl "info"
      assert.equal logLvl, logger.Info
      # TODO: for each possible log level ...
      logLvl = logObj._validateLogLvl "critical"
      assert.equal logLvl, logger.Critical

  it 'sets level to Debug, if given an invalid log level', ->
      logLvl = logObj._validateLogLvl "sometest"
      assert.equal logLvl, logger.Debug
```